src/controllers/oauthContoller.ts

```diff
@@ -131,6 +132,44 @@ export const getClaims = async (
   }
 };
 
+export const verifyPKCE = async (
+  req: Request,
+  res: Response,
+  next: NextFunction
+) => {
+  // getting cliams from code before revoke authorization code, for id_token
+  const { code } = req.body;
+  if (!code) {
+    next();
+    return;
+  }
+  try {
+    const codeData = await prisma.code.findUnique({
+      where: {
+        code: code
+      },
+      select: {
+        codeChallenge: true
+      }
+    });
+    if (codeData?.codeChallenge) {
+      if (!req.body.code_verifier) {
+        return res.status(400).json({ message: 'code_verifier is missing' });
+      }
+      const isVerified = await verifyCodeChallenge(
+        code,
+        req.body.code_verifier
+      );
+      if (!isVerified) {
+        return res.status(400).json({ message: 'Internal server error.' });
```

Comment:
return proper error if verification fails, check statandard.

as i can see `verifyCodeChallenge` return bool and throws error.
return `internal server error` if verifyCodeCahellenge throws an error.